ledger-tool/src/output.rs

```diff
@@ -637,6 +648,33 @@ impl AccountsScanner {
             && (self.config.include_sysvars || !solana_sdk::sysvar::is_sysvar_id(pubkey))
     }
 
+    fn maybe_output_account<S>(
+        &self,
+        seq_serializer: &mut Option<S>,
+        pubkey: &Pubkey,
+        account: &AccountSharedData,
+        slot: Option<Slot>,
+        cli_account_new_config: &CliAccountNewConfig,
+    ) where
+        S: SerializeSeq,
+    {
+        if self.config.include_account_contents {
+            if let Some(serializer) = seq_serializer {
+                let cli_account =
+                    CliAccount::new_with_config(pubkey, account, cli_account_new_config);
+                serializer.serialize_element(&cli_account).unwrap();
+            } else {
+                output_account(
+                    pubkey,
+                    account,
+                    slot,
+                    self.config.include_account_data,
+                    self.config.account_data_encoding,
+                );
+            }
+        }
+    }
```

Comment:
Unrelated to this PR, as you are just moving this code.
But I find it a bit confusing that this function behaves so differently depending on if it was provided with a serializer or not.

I wonder if it would have a more straightforward to always require a serializer here.
But then the caller could call it with a serializer that turns into an `output_account()` call, in cases when they want the output to be printed using the `output_account()`.

`CliAccountNewConfig` also looks a bit suspicious here, as it is only used in one case and not the other.

Reply:
This is a result of not wanting to duplicate code. In a previous round of cleanup, I condensed some stuff into helpers so the duplicate code wouldn't be as bad now if we moved the "if" statement that chooses between print/json higher (and have a separate function each for print/json). For the sake of this PR, I'm inclined to leave things as-is

Reply:
Absolutely, I was not suggesting you make any changes to this part in this PR.
Was just wondering about it, as it was part of the patch :)